### PR TITLE
[Mandatory 2FA] occ commands revamp

### PR DESCRIPTION
**Pull request by ChristophWurst**
Ref https://github.com/nextcloud/server/issues/11019.

ToDo
- [x] Refactor enable/disable commands
- [x] Update provider registry when providers get enabled/disabled
- [x] Add cleanup command
- [x] Test with providers that implement either of the interfaces
- [x] ~~Update documentation~~ -> tracked at https://github.com/orgs/nextcloud/projects/17#card-13042440
- [x] Requires #11139 
- [x] Tests

### CODE REVIEW

File: core/Command/TwoFactorAuth/Disable.php [hunk 1]
```diff
@@ -24,21 +24,22 @@
...
+	/** @var ProviderManager */
 	private $manager;
 
 	/** @var IUserManager */
 	protected $userManager;
 
-	public function __construct(Manager $manager, IUserManager $userManager) {
-		parent::__construct('twofactorauth:disable');
+	public function __construct(ProviderManager $manager, IUserManager $userManager) {
+		parent::__construct('twofactorauth:enable');
```
Comment (danielkesselberg) [inline core/Command/TwoFactorAuth/Disable.php]: I think this should be "disable"? But setName is called in configure anyway so you could remove it here.

Comment (ChristophWurst) [reply core/Command/TwoFactorAuth/Disable.php]: Good catch, fixed!

**Review by blizzz — APPROVED** [review]
looks good :elephant:

**Review by rullzer — APPROVED** [review]
Looking good :rocket:

**Pull request merged**